packages/legend-extension-dsl-data-space/src/graph-manager/protocol/pure/DSL_DataSpace_PureProtocolProcessorPlugin.ts

```diff
@@ -353,14 +385,37 @@ export class DSL_DataSpace_PureProtocolProcessorPlugin
             return elementPointer;
           });
           protocol.executables = metamodel.executables?.map((executable) => {
-            const executableProtocol = new V1_DataSpaceExecutable();
-            executableProtocol.title = executable.title;
-            executableProtocol.description = executable.description;
-            executableProtocol.executable = new V1_PackageableElementPointer(
-              undefined,
-              executable.executable.valueForSerialization ?? '',
-            );
-            return executableProtocol;
+            if (executable instanceof DataSpaceExecutableTemplate) {
+              const executableProtocol = new V1_DataSpaceTemplateExecutable();
+              executableProtocol.title = executable.title;
+              executableProtocol.description = executable.description;
+              if (executable.executionContextKey) {
+                executableProtocol.executionContextKey =
+                  executable.executionContextKey;
+              }
+              executableProtocol.query =
+                executable.query.accept_RawValueSpecificationVisitor(
+                  new V1_RawValueSpecificationTransformer(context),
+                ) as V1_RawLambda;
+              return executableProtocol;
+            } else if (
+              executable instanceof DataSpacePackageableElementExecutable
+            ) {
+              const executableProtocol =
+                new V1_DataSpacePackageableElementExecutable();
+              executableProtocol.title = executable.title;
+              executableProtocol.description = executable.description;
+              executableProtocol.executable = new V1_PackageableElementPointer(
+                undefined,
+                executable.executable.valueForSerialization ?? '',
+              );
+              return executableProtocol;
+            } else {
+              throw new UnsupportedOperationError(
```

Comment:
Same here as above. Did we have any cases before where the executable wasn't an instance of `DataSpacePackageableElementExecutable` but we still built the `V1_DataSpaceExecutable` object instead of throwing an error? Would this change break any of those cases?